ruby/ql/lib/codeql/ruby/frameworks/core/Kernel.qll

```diff
@@ -18,17 +18,34 @@ module Kernel {
    * providing a specific receiver as in `Kernel.exit`.
    */
   class KernelMethodCall extends DataFlow::CallNode {
+    string methodName;
+
     KernelMethodCall() {
-      this = API::getTopLevelMember("Kernel").getAMethodCall(_)
+      this = API::getTopLevelMember("Kernel").getAMethodCall(methodName)
       or
       this.asExpr().getExpr() instanceof UnknownMethodCall and
+      (
+        methodName = super.getMethodName()
+        or
+        this.asExpr().getExpr() instanceof SuperCall and
+        methodName = this.asExpr().getExpr().getEnclosingCallable().(MethodBase).getName()
```

Comment:
I think it would make sense if super calls always report the name of its enclosing method as its name. Library models shouldn't have to bother with the kind of code we see here - we should take it as an opportunity to improve the API.

Reply:
That sounds like a good idea, I'll try it and see what happens.

Reply:
I didn't really have to change the QLDoc for `getMethodName()` as it already said: `Gets the name of the method being called.`.  
And the method being called is not `super`, so one could say that this new implementation actually fixes a bug.

Reply:
Sounds like a good idea to me.